packages/svelte/src/compiler/migrate/index.js

```diff
@@ -729,9 +744,43 @@ const template = {
 		}
 		next();
 	},
+	SvelteSelf(node, { state, next }) {
+		const source = state.str.original.substring(node.start, node.end);
+		if (!state.filename) {
+			const indent = guess_indent(source);
+			state.str.prependRight(
+				node.start,
+				`<!-- @migration-task: svelte:self is deprecated, import this Svelte file into itself instead -->\n${indent}`
+			);
+			next();
+			return;
+		}
+		// overwrite the open tag
+		state.str.overwrite(
+			node.start + 1,
+			node.start + 1 + 'svelte:self'.length,
+			`${state.names.svelte_self}`
+		);
+		// if it has a fragment we need to overwrite the closing tag too
+		if (node.fragment.nodes.length > 0) {
+			const start_closing =
+				state.str.original.indexOf('<', node.fragment.nodes[node.fragment.nodes.length - 1].end) +
```

Comment:
You can also do `lastIndexOf('<', node.end)`

Reply:
Stop being so smart! Joking, please continue 😄